packages/web-frontend/src/containers/Map/Map.js

```diff
@@ -0,0 +1,289 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+import { createSelector } from 'reselect';
+import { TileLayer, MarkerLayer, LeafletMap } from '@tupaia/ui-components/lib/map';
+import { checkBoundsDifference, organisationUnitIsArea } from '../../utils';
+import { DemoLand } from './DemoLand';
+import { ConnectedPolygon } from './ConnectedPolygon';
+import { DisasterLayer } from './DisasterLayer';
+import { ZoomControl } from './ZoomControl';
+import {
+  selectActiveTileSet,
+  selectAllMeasuresWithDisplayInfo,
+  selectCurrentOrgUnit,
+  selectHasPolygonMeasure,
+  selectOrgUnit,
+  selectOrgUnitChildren,
+  selectOrgUnitSiblings,
+  selectRenderedMeasuresWithDisplayInfo,
+} from '../../selectors';
+import { changePosition, closeDropdownOverlays, setOrgUnit } from '../../actions';
+import { TRANS_BLACK, TRANS_BLACK_LESS } from '../../styles';
+
+const CHANGE_TO_PARENT_PERCENTAGE = 0.6;
+
+const StyledMap = styled(LeafletMap)`
+  height: 100vh;
+  width: 100%;
+
+  .leaflet-control-zoom {
+    z-index: 1;
+    border: none;
+    top: -50px;
+    right: 350px;
+    transition: right 0.5s ease;
+
+    a {
+      background: ${TRANS_BLACK_LESS};
+      box-shadow: none;
+      border: none;
+      color: white;
+
+      &:hover {
+        background: ${TRANS_BLACK};
+        box-shadow: none;
+      }
+    }
+  }
+`;
+
+/**
+ * Map
+ * Includes basic map setup/rendering, controlled through props that are connected to the redux store.
+ * Rendering includes heatmaps markers, polygons.
+ */
+class MapComponent extends Component {
+  componentWillMount() {
+    window.addEventListener('resize', () => this.forceUpdate());
+  }
+
+  shouldComponentUpdate(nextProps) {
+    const {
+      currentOrganisationUnit,
+      displayedChildren,
+      measureInfo,
+      measureData,
+      position,
+      tileSetUrl,
+      sidePanelWidth,
+    } = this.props;
+    // Only updates/re-renders when the measure has changed or the orgUnit has changed.
+    // These are the only cases where polygons or area tooltips should rerender.
+    if (nextProps.measureInfo.measureId !== measureInfo.measureId) return true;
+
+    if (JSON.stringify(nextProps.displayedChildren) !== JSON.stringify(displayedChildren))
+      return true;
```

Comment:
I checked an it's not a rule. Interestingly, how this could come about is if you have a conditional on a single line and then the line gets too long, prettier will move the return down. I am personally not a big fan of the single line conditional expressions as I find the inconsistency harder to read.

Reply:
Yeah I'm with you on that